### PR TITLE
Fixes freezes system with Fog Sun Scatter property on Windows 10 (QEMU)

### LINKED ISSUE #79509: Using Fog Sun Scatter property on Windows 10 (QEMU) freezes system
### Godot version

v4.1-stable (latest from website)

### System information

Windows 10 (QEMU) Godot_v4.1-stable_win64 Forward+ Vulkan

### Issue description

I am using Shadow Cloud for development which is done using QEMU virtualization, so it's may be a problem with hardware virtualization

I can't capture video due to OBS crash after unfreezing

After 3 minute frezze i got this errors in console.

### Steps to reproduce

1. Create WorldEnvironment
2. Enable fog
3. Set any value to Sun Scatter property

### Minimal reproduction project

Blank project with 3D scene and WorldEnvironment

### PR DESCRIPTION
**Pull request by anuragdaksh7**
* *Bugsquad edit, fixes: #79509*

*Bugsquad edit, alternative/complementary to: #79812*

### CODE REVIEW

**Comment by AThousandShips** [comment]
Please provide a more descriptive name for this PR and for the commit message

File: servers/rendering/renderer_rd/environment/sky.cpp [hunk 1]
```diff
@@ -1164,6 +1111,62 @@ void SkyRD::setup_sky(RID p_env, Ref<RenderSceneBuffersRD> p_render_buffers, con
...
+
+		ERR_FAIL_COND(!material);
+
+		shader_data = material->shader_data;
+
+		ERR_FAIL_COND(!shader_data);
+
+		material->set_as_used();
+
+		// Save our screen size, our buffers will already have been cleared
```
Comment (AThousandShips) [inline servers/rendering/renderer_rd/environment/sky.cpp:1137]:
```suggestion
		// Save our screen size, our buffers will already have been cleared.
```
Code style requires that comments begin with a capital and end with a period, same for other comments here

File: servers/rendering/renderer_rd/environment/sky.cpp [hunk 2]
```diff
@@ -1164,6 +1111,62 @@ void SkyRD::setup_sky(RID p_env, Ref<RenderSceneBuffersRD> p_render_buffers, con
...
+			material->uniform_set_updated = false;
+			sky->reflection.dirty = true;
+		}
+
+		if (!p_cam_transform.origin.is_equal_approx(sky->prev_position) && shader_data->uses_position) {
+			sky->prev_position = p_cam_transform.origin;
+			sky->reflection.dirty = true;
+		}
+
+
```
Comment (AThousandShips) [inline servers/rendering/renderer_rd/environment/sky.cpp:1168]: Remove these two empty lines

Comment (anuragdaksh7) [reply servers/rendering/renderer_rd/environment/sky.cpp:1168]: ok

File: servers/rendering/renderer_rd/environment/sky.cpp [hunk 3]
```diff
@@ -1012,60 +1012,7 @@ void SkyRD::setup_sky(RID p_env, Ref<RenderSceneBuffersRD> p_render_buffers, con
...
-			material->uniform_set_updated = false;
-			sky->reflection.dirty = true;
-		}
-
-		if (!p_cam_transform.origin.is_equal_approx(sky->prev_position) && shader_data->uses_position) {
-			sky->prev_position = p_cam_transform.origin;
-			sky->reflection.dirty = true;
-		}
-
+	if (sky ||render_fog){
```
Comment (AThousandShips) [inline servers/rendering/renderer_rd/environment/sky.cpp:1015]:
```suggestion
	if (sky || render_fog) {
```

File: servers/rendering/renderer_rd/environment/sky.cpp [hunk 4]
```diff
@@ -1164,6 +1111,60 @@ void SkyRD::setup_sky(RID p_env, Ref<RenderSceneBuffersRD> p_render_buffers, con
...
+
+		ERR_FAIL_COND(!shader_data);
+
+		material->set_as_used();
+
+		// Save our screen size, our buffers will already have been cleared.
+		sky->screen_size.x = p_screen_size.x < 4 ? 4 : p_screen_size.x;
+		sky->screen_size.y = p_screen_size.y < 4 ? 4 : p_screen_size.y;
+
+		// Trigger updating radiance buffers
```
Comment (AThousandShips) [inline servers/rendering/renderer_rd/environment/sky.cpp:1141]:
```suggestion
		// Trigger updating radiance buffers.
```

**Comment by AThousandShips** [comment]
Once you've added all the changes please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Review by AThousandShips — CHANGES_REQUESTED** [review]
This unfortunately won't work, it moves the initialisation of `shader_data` after it is used

**Pull request closed without merging**